pkg/binary/envoy/debug/pii.go

```diff
@@ -0,0 +1,72 @@
+package debug
+
+import (
+	"fmt"
+	"bitbucket.org/creachadair/shell"
+	"crypto/sha256"
+)
+
+// filter the array of logs using formatStr and validFieldNames, hash the resulting array of logs and return the final array
+func process(logsArr []string, formatStr string, validFieldNames map[string]bool) (processedLogs []string) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("process func panicking, recovering")
+		}
+	} ()
+
+	fieldNames, ok := shell.Split(formatStr)
+
+	if !ok {
+		panic(fmt.Sprint("error in splitting format string: %s", formatStr))
+	}
+
+	testForValidLog :=  func(log string) bool {
+		fields, ok := shell.Split(log)
+		if !ok {
+			panic(fmt.Sprint("error in splitting log: %s", log))
+		}
+		// TODO: consider using regex to check fields instead of merely verifiying the number of fields
+		return len(fields) == len(fieldNames)
+	}
+	validLogs := filter(logsArr, testForValidLog)
+
+	// filter and hash each log in the array of valid logs with relevant fields
+	for i,_ := range(validLogs) {
+		fields, ok := shell.Split(validLogs[i])
```

Comment:
rather than splitting once when we call `filter` then a second time when we process them here, let's just remove the filter and check that the log line is valid in this single loop: in this loop, just `parts, ok := shell.Split(log)` and `if !ok || !valid(fieldNames, parts) { log.debugf(...); continue }`. Remember that the number of logs can be very large, so we don't want to touch each log line more than once if we can avoid it.